### PR TITLE
Answer:12. RxAngular solution

### PR DESCRIPTION
**Pull request by Karnaukhov-kh**
## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: apps/scroll-cd/src/app/listen-outside-zone.util.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+): () => void {
+  getZoneUnPatchedApi(target, 'addEventListener').call(target, type, listener);
+  return () => {
+    getZoneUnPatchedApi(target, 'removeEventListener').call(
+      target,
+      type,
+      listener
+    );
+  };
+}
```
Comment (tomalaforge) [inline apps/scroll-cd/src/app/listen-outside-zone.util.ts:16]: question: why do you need to call getZoneUnPatchedApi twice ?

**Comment by tomalaforge** [comment]
Thanks for submitting. 

Really neat solution. RxAngular is very powerful to do such thing.

File: apps/scroll-cd/src/app/listen-outside-zone.util.ts (hunk 1 above)
Comment (Karnaukhov-kh) [reply apps/scroll-cd/src/app/listen-outside-zone.util.ts:16]: Thanks for accepting the solution :). `removeEventListener` is also patched by zone. So if you want to go all-in zone-less it's better to unpatch removal also.

Native 

remove-patched

Unpatched

remove-unpatched

Comment (tomalaforge) [reply apps/scroll-cd/src/app/listen-outside-zone.util.ts:16]: I'm so tired, I didn't see that this was two different listeners 😅

**Pull request closed without merging**